fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java

```diff
@@ -166,75 +213,122 @@ public IDatabaseTranslator getTranslator() {
-        try {
-            Connection c = getConnection();
-            try {
-                JdbcTarget target = new JdbcTarget(c);
-                DerbyAdapter adapter = new DerbyAdapter(target);
 
-                // Replace the target with a decorated output, so that we print all the DDL before executing
-                // The output is very FINE and logs out a lot. 
-                if (logger.isLoggable(Level.FINE)) {
-                    PrintTarget printer = new PrintTarget(target, logger.isLoggable(Level.FINE));
-                    adapter = new DerbyAdapter(printer);
-                }
-                fn.accept(adapter);
-            } catch (DataAccessException x) {
-                logger.log(Level.SEVERE, "Error while running", x);
-                c.rollback();
-                throw x;
-            }
-            c.commit();
-        } catch (SQLException e) {
-            logger.log(Level.SEVERE, "Error while running", e);
-            throw DERBY_TRANSLATOR.translate(e);
-        } finally {
-            logger.info("connection was closed");
+        IConnectionProvider cp = new DerbyConnectionProvider(this, null);
+        ConnectionProviderTarget target = new ConnectionProviderTarget(cp);
+        DerbyAdapter adapter = new DerbyAdapter(target);
+
+        // Replace the target with a decorated output, so that we print all the DDL before executing
+        // The output is very FINE and logs out a lot. 
+        if (logger.isLoggable(Level.FINE)) {
+            PrintTarget printer = new PrintTarget(target, logger.isLoggable(Level.FINE));
+            adapter = new DerbyAdapter(printer);
         }
+      
+        // call the Function we've been given using the adapter we just wrapped
+        // around the connection. Each statement executes in its own connection/transaction.
+        fn.accept(adapter);
     }
 
+    /**
+     * Diagnostic utility to display all the current locks in the Derby database
+     */
+    public void dumpLockInfo() {
+        DerbyLockDiag diag = new DerbyLockDiag();
+        IConnectionProvider cp = new DerbyConnectionProvider(this, null);
+        ConnectionProviderTarget target = new ConnectionProviderTarget(cp);
+        DerbyAdapter adapter = new DerbyAdapter(target);
+        List<LockInfo> locks = adapter.runStatement(diag);
+        System.out.println(LockInfo.header());
+        locks.forEach(System.out::println);
+    }
+    
+    /**
+     * Dump locks using the given connection
+     * @param c
+     */
+    public static void dumpLockInfo(Connection c) {
+        // wrap the connection so that we can run our lock diag DAO
+        JdbcTarget target = new JdbcTarget(c);
+        DerbyAdapter adapter = new DerbyAdapter(target);
+        
+        DerbyLockDiag diag = new DerbyLockDiag();
+        List<LockInfo> locks = adapter.runStatement(diag);
+        
+        // render
+        System.out.println(LockInfo.header());
```

Comment:
Fix the logger

Reply:
I don't want to use a logger in this case.